SU2_CFD/src/solver_direct_heat.cpp

```diff
@@ -1460,7 +1464,10 @@ void CHeatSolverFVM::SetTime_Step(CGeometry *geometry, CSolver **solver_containe
 
     thermal_diffusivity = config->GetThermalDiffusivity_Solid();
     if(flow) {
-      eddy_viscosity = solver_container[FLOW_SOL]->node[iPoint]->GetEddyViscosity();
+      if(turb) {
+        eddy_viscosity = solver_container[TURB_SOL]->node[iPoint]->GetmuT();
+      }
+
       thermal_diffusivity = laminar_viscosity/Prandtl_Lam + eddy_viscosity/Prandtl_Turb;
```

Comment:
Once more of the driver changes are implemented, if you agree, I think we can eliminate the weakly_coupled mode for the heat solver, leaving just the native incompressible energy eqn., which would reduce code complexity quite a bit (the type of check for diffusivity here, the convective schemes in the heat solver and other terms, plus many of the weakly_coupled boolean checks throughout other classes like output). Maybe you already have plans for this or have other thoughts?

Reply:
Yes, all these weakly-heat-booleans to not contribute to the overall code reading convenience, I totally agree. Sorry for that.

Still there is one good reason why one could keep it (besides it's helping me a lot to add my heat-related contributions at the moment), which is the possibility to easily transform **weakly_coupled_heat** into **passive_scalar**, and then having the adjoint functionality directly available.

(As the heat solver itself is just an advection-diffusion-solver which can take any form you like, it would be just some preprocessing turning it into a heat, Poisson or whatever solver. Implementation-wise, one would replace the overly specific variable names like thermal_conductivity simply into constant_diffusion or similar.)
I will make it one side-project of mine after the multizone adjoints will have been brought to life.